### PR TITLE
Add Enhanced Documentation

### PR DESCRIPTION
**Pull request by nuttercd**
Tell me what to fix please.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,3 +18,15 @@ This is a machine code simulator coded in Python 3 based upon the
 |A         | R*X     | ROTATE the bit pattern in register R one bit to the right X amount of times. Each time placing the bit that started on the low end on the high end.
 |B         | RXY     | JUMP to the instruction located in the memory cell at address XY if the bit pattern in register R is equal to the bit pattern in register number 0. Otherwise, continue with the normal sequence of execution. (The jump is implemented by copying XY into the program counter during the execute phase.) |
 |C         | ***     | HALT |
+
+
+## How it works
+Upon execution of the program the user is prompted with the option of choosing the number of register
+cells and the number of Memory cells they would like. The user may choose between 1-16 register cells
```
Comment (showalter) [inline README.md]: Don't capitalize memory

File: README.md [hunk 2]
```diff
@@ -18,3 +18,15 @@ This is a machine code simulator coded in Python 3 based upon the
...
+and 0-255 memory cells.
```
Comment (showalter) [inline README.md]: 0 cells isn't an option, and 256 is.

File: README.md [hunk 3]
```diff
@@ -18,3 +18,15 @@ This is a machine code simulator coded in Python 3 based upon the
...
+
+After the user's cells are chosen the user may edit the contents of the register cell with "R"
```
Comment (showalter) [inline README.md]: This is a great place for a list.

File: __main__.py [hunk 4]
```diff
@@ -44,54 +50,88 @@ def execute_instruction(a, b):
...
+        str: The complete instruction
+    """
     completeinstruction = (a + b)
 
     return completeinstruction
 
 
-# If instruction looks like 1RXY, load register R
-# with the bits found in memory cell XY
+
```
Comment (showalter) [inline __main__.py]: There should be two lines between functions rather than three

File: __main__.py [hunk 5]
```diff
@@ -44,54 +50,88 @@ def execute_instruction(a, b):
...
 def load_from_cell(instruction):
+    """ If instruction looks like 1RXY
+    load register R with the bits found in memory cell XY
+
+    Args:
+        instruction (str):
```
Comment (showalter) [inline __main__.py]: Add a description

File: display.py [hunk 6]
```diff
@@ -1,4 +1,14 @@
 def describe(a, b):
+    """Displays an explanation of the instruction
+
+    Args:
+        a (str): The string of the first half of the description
+        b (str): The string of the second half of the description
+
+    Returns:
+        The complete description
```
Comment (showalter) [inline display.py]: type?

File: processor.py [hunk 7]
```diff
@@ -6,13 +6,18 @@ def __init__(self, number):
...
+        """Values must be passed as a string representation of a hex value
+        or as a hex value. They should not be passed as ints
+
+        """
         self.value = int(value, 16)
 
-    # Values will be returned as a base 16 int.
+
     def getvalue(self):
+        """Values will be returned as base 16 int.
```
Comment (showalter) [inline processor.py]: There's a return here

File: processor.py [hunk 8]
```diff
@@ -6,13 +6,18 @@ def __init__(self, number):
     def getid(self):
         return self.number[2:]
 
-    # Values must be passed as a string representation of a hex
-    # value, or as a hex value. They should not be passed as ints.
     def setvalue(self, value):
+        """Values must be passed as a string representation of a hex value
+        or as a hex value. They should not be passed as ints
+
```
Comment (showalter) [inline processor.py:32]: There should be a parameter here.

**Review by showalter — CHANGES_REQUESTED** [review]

**Review by showalter — CHANGES_REQUESTED** [review]
Docstrings for each module and the cell class should be added

File: __main__.py [hunk 9]
```diff
@@ -44,54 +50,87 @@ def execute_instruction(a, b):
...
 
 
-# If instruction looks like 2RXY, load register R
-# with the bit pattern XY
 def load_with(instruction):
+    """If instruction looks like 2RXY
+    LOAD register R with the bit pattern XY
+
+    Args:
+        instruction (str): the instruction being executed in this case starts with a 2
```
Comment (showalter) [inline __main__.py]: This is awkward working

File: README.md [hunk 10]
```diff
@@ -2,8 +2,32 @@
...
+After configuration, the user may use the following keys to edit the cells.
+
+- `r` - edit the contents of register cells
+- `m` - edit the contents of memory cells
+- `return` - step through the program
+- `i` - edit the instruction counter
+- `e` - execute the program
+- Any other input will quit the program
+
+### Language Description Table
```
Comment (showalter) [inline README.md:30]: We should really clarify that opcode 6 does the same thing as 5 for now.

File: README.md [hunk 11]
```diff
@@ -2,19 +2,49 @@
...
 |Op-Code   | Operand | Description |
 |:--------:|:-------:|-------------|
 |1         | RXY     | LOAD register R with the bits found in memory cell XY |
 |2         | RXY     | LOAD register R with the hex value XY|
 |3         | RXY     | STORE the contents found in register R in the memory cell XY |
 |4         | *RS     | MOVE the bit pattern found in register R to register S | 
-|5         | RST     | ADD the bit patterns in registers S and T and put the result in register R in a two’s complement complement representation |
-|6         | RST     | ADD the bit patterns in registers S and T and put the result in register R in a floating-point notation |
+|5         | RST     | ADD the bit patterns in registers S and T and put the result in register R in a two’s complement complement representation|
+|6         | RST     | ADD the bit patterns in registers S and T and put the result in register R in a floating-point notation (**NOTE** Op-Code 6 does the same as 5)|
```
Comment (showalter) [inline README.md]: clarify that for now, it is the same operation, and that floating point addition is not available.

File: processor.py [hunk 12]
```diff
@@ -6,13 +6,24 @@ def __init__(self, number):
...
+        """
         self.value = int(value, 16)
 
-    # Values will be returned as a base 16 int.
+
     def getvalue(self):
+        """Values will be returned as base 16 int.
+
+        Return:
```
Comment (showalter) [inline processor.py]: returns

File: processor.py [hunk 13]
```diff
@@ -6,13 +6,24 @@ def __init__(self, number):
...
+            int: an integer representation of the cells contents
```
Comment (showalter) [inline processor.py]: cell's

File: processor.py [hunk 14]
```diff
@@ -6,13 +6,24 @@ def __init__(self, number):
...
+        Args:
+            value (str): a hex representation of the contents of the cell
+
+        """
         self.value = int(value, 16)
 
-    # Values will be returned as a base 16 int.
+
     def getvalue(self):
+        """Values will be returned as base 16 int.
```
Comment (showalter) [inline processor.py]: This should be a description, not just a technicality on how to use the function.

File: processor.py [hunk 15]
```diff
@@ -6,13 +6,24 @@ def __init__(self, number):
     def getid(self):
         return self.number[2:]
 
-    # Values must be passed as a string representation of a hex
-    # value, or as a hex value. They should not be passed as ints.
     def setvalue(self, value):
+        """Values must be passed as a string representation of a hex value
```
Comment (showalter) [inline processor.py]: This should be a description

File: processor.py (hunk 13 above)
Comment (nuttercd) [reply processor.py]: what do you mean by cells???

File: __main__.py [hunk 16]
```diff
@@ -44,54 +50,87 @@ def execute_instruction(a, b):
...
 
 
-# If instruction looks like 1RXY, load register R
-# with the bits found in memory cell XY
 def load_from_cell(instruction):
+    """ If instruction looks like 1RXY
+    load register R with the bits found in memory cell XY
+
+    Args:
+        instruction (str): instruction (str): the instruction being executed, in this case it starts with a 1
```
Comment (showalter) [inline __main__.py]: delete duplicate

File: processor.py [hunk 17]
```diff
@@ -6,13 +6,25 @@ def __init__(self, number):
     def getid(self):
         return self.number[2:]
 
-    # Values must be passed as a string representation of a hex
-    # value, or as a hex value. They should not be passed as ints.
     def setvalue(self, value):
+        """Sets the contents of the cell.
+        (The value must be passed as a hexadecimal value or as a string
```
Comment (showalter) [inline processor.py]: Parentheses should not be used here, and I'd suggest adding a newline between the description and the extra information.

**Review by showalter — CHANGES_REQUESTED** [review]
Most of this looks good so far, but I'd like for each module and the Cell class to have a docstring as well.

File: __main__.py [hunk 18]
```diff
@@ -1,3 +1,8 @@
+"""The main file contains the main function and other functions for constructing cells and executing machine code
```
Comment (showalter) [inline __main__.py]: Each module string should have a little more substance than this. A copyright notice should be here. Also, module attributes, such as registers, cells, and icounter, should have descriptions in the module docstring or directly below each of those attributes.

File: __main__.py [hunk 19]
```diff
@@ -44,54 +55,87 @@ def execute_instruction(a, b):
...
     xy = instruction[2:]
     xy = cells[int(xy, 16)]
 
     r.setvalue(hex(xy.getvalue()))
 
 
-# If instruction looks like 2RXY, load register R
-# with the bit pattern XY
 def load_with(instruction):
+    """If instruction looks like 2RXY
```
Comment (showalter) [inline __main__.py]: Can we move each of the "if instruction looks like ..." to a different line?
Example:
```
Load a register R with the bit pattern XY

Instruction should look like (opcode)RXY
...
```

If we want to change the opcodes later, it's easier to just say (opcode) now than changing all of the comments later.

File: __main__.py [hunk 20]
```diff
@@ -193,6 +272,12 @@ def execute(step):
 
 
 def main():
+    """ The main function constructs memory and register cells and allows the user to edit them
+     and executes the contents of the cells
+
+     """
+
+
```
Comment (showalter) [inline __main__.py:304]: only have one empty line

File: __main__.py [hunk 21]
```diff
@@ -193,6 +272,12 @@ def execute(step):
 
 
 def main():
+    """ The main function constructs memory and register cells and allows the user to edit them
+     and executes the contents of the cells
+
```
Comment (showalter) [inline __main__.py]: I think you can move the ending quotes up here

File: processor.py [hunk 22]
```diff
@@ -1,22 +1,57 @@
...
     def __init__(self, number):
+        """Constructs and intializes the id and value for a cell
+
+        """
         self.number = number
         self.value = None
 
+
     def getid(self):
+        """Returns the string representation of the ID
```
Comment (showalter) [inline processor.py]: Indentation and whitespace

File: processor.py [hunk 23]
```diff
@@ -1,22 +1,57 @@
...
+
+    Attributes:
+        id (int): the number of the cell
+        value (int): the contents of the cell
+
+    """
+
+
     def __init__(self, number):
+        """Constructs and intializes the id and value for a cell
```
Comment (showalter) [inline processor.py]: whitespace

File: processor.py [hunk 24]
```diff
@@ -1,22 +1,57 @@
...
+        Returns:
+            int: an integer representation of the cell's contents
+
+        """
         if self.value is not None:
             return self.value
 
+
     def tostr(self):
+        """Returns the a string reorientation of the contents of the cell.
```
Comment (showalter) [inline processor.py]: whitespace

**Review by showalter — CHANGES_REQUESTED** [review]

**Review by nuttercd — COMMENTED** [review]
These changes are good.

**Review by showalter — APPROVED** [review]

**Pull request merged**